doc/source/user/absolute_beginners.rst

```diff
@@ -5,212 +5,199 @@ NumPy: the absolute basics for beginners
 
 Reading the example code
 ------------------------
 
-If you aren't already comfortable with reading tutorials that contain a lot of code,
-you might not know how to interpret a code block that looks
-like this::
-
-  >>> a = np.arange(6)
-  >>> a2 = a[np.newaxis, :]
-  >>> a2.shape
-  (1, 6)
+Throughout the NumPy documentation, you will find blocks that look like::
 
-If you aren't familiar with this style, it's very easy to understand.
-If you see ``>>>``, you're looking at **input**, or the code that
-you would enter. Everything that doesn't have ``>>>`` in front of it
-is **output**, or the results of running your code. This is the style
-you see when you run ``python`` on the command line, but if you're using
-IPython, you might see a different style. Note that it is not part of the
-code and will cause an error if typed or pasted into the Python
-shell. It can be safely typed or pasted into the IPython shell; the ``>>>``
-is ignored.
+  >>> a = np.array([[1, 2, 3],
+  ...               [4, 5, 6]])
+  >>> a.shape
+  (2, 3)
 
+Text preceded by ``>>>`` or ``...`` is **input**, the code that you would
+enter in a script or at a Python prompt. Everything else is **output**, the
+results of running your code. Note that ``>>>`` and ``...`` are not part of the
+code and may cause an error if entered at a Python prompt.
 
 What’s the difference between a Python list and a NumPy array?
 --------------------------------------------------------------
 
-NumPy gives you an enormous range of fast and efficient ways of creating arrays
-and manipulating numerical data inside them. While a Python list can contain
-different data types within a single list, all of the elements in a NumPy array
-should be homogeneous. The mathematical operations that are meant to be performed
-on arrays would be extremely inefficient if the arrays weren't homogeneous.
+Python lists are excellent, general-purpose containers. They can be
+"heterogeneous", meaning that they can contain elements of a variety of types,
+and they are quite fast when used to perform individual operations on a handful
+of elements.
 
-**Why use NumPy?**
+Depending on the characteristics of the data and the types of operations that
+need to be performed, other containers may be more appropriate; by exploiting
+these characterists, we can improve speed, reduce memory consumption, and
+offer a high-level syntax for performing a variety of common processing tasks.
+NumPy shines when there are large quantities of "homogeneous" (same-type) data
+to be processed on the CPU.
 
-NumPy arrays are faster and more compact than Python lists. An array consumes
-less memory and is convenient to use. NumPy uses much less memory to store data
-and it provides a mechanism of specifying the data types. This allows the code
-to be optimized even further.
+What is an "array"?
+-------------------
 
-What is an array?
------------------
+An "array" is an instance of the fundamental data structure class of the NumPy
```

Comment:
"data structure class" is a little awkward. Also an array is a general programming concept, so I'd first describe that then explain how numpy implements the array data structure in `ndarray`. Maybe something like:

> Arrays are a data structure that many programming languages and environments provide as a fundamental abstraction for storing an ordered collection of instances of a single type of data. 
>
> NumPy provides the `ndarray` class, to represent an N-dimensional array of data. 
>
>A one-dimensional array is just like a Python list, it holds an ordered collection of items with a single integer index. A 2-dimensional array is a bit like a spreadsheet, but with a fixed number of cells that can only hold values with one data type. NumPy also generalizes to an arbitrary number of dimensions.

Or maybe move the discussion of the name until below where you introduce higher dimensional arrays, if you don't want to talk about that yet. I do like the spreadsheet analogy for beginners though.

Reply:
I agree. This was a quick attempt to address https://github.com/numpy/numpy/pull/24981#discussion_r1370087499; see diff of latest commit. I'll think more carefully about it!

Reply:
The first sentence serves as a general definition of an array. You listed two characteristics that distinguish an array from other data structures:
- ordered
- homogeneous

Below that, you also gave the example of a fixed height/width spreadsheet. This made me think of some two other characteristics that might belong in the definition:
- fixed size
- contiguous in memory

Another option is *not* giving all of those properties, but stating that an array supports efficient performance of a common operation on many elements at once (i.e. array programming). 

With these other possibilities in mind, what do you think belongs in the definition?

Reply:
How about:

```
In computer programming, an array is a structure for storing and retrieving
data. We often talk about an array as if it were a grid in space, with each
cell storing one element of the data. For instance, if each element of the
data were a number, we might visualize a "one-dimensional" array like a
list:

+---+---+---+---+
| 1 | 5 | 2 | 0 |
+---+---+---+---+

A two-dimensional array would be like a table:

+---+---+---+---+
| 1 | 5 | 2 | 0 |
+---+---+---+---+
| 8 | 3 | 6 | 1 |
+---+---+---+---+
| 1 | 7 | 2 | 9 |
+---+---+---+---+

A three-dimensional array would be like a set of tables, perhaps stacked
as though they were printed on separate pages. In NumPy, this idea is
generalized to an arbitrary number of dimensions, and so the fundamental
array class is called ``ndarray``: it represents an "N-dimensional
array".

Most NumPy arrays have some restrictions. For instance:

- All elements of the array must be of the same type of data
- Once created, the total size of the the array can't change.
- The shape must be "rectangular", not "jagged"; e.g., each row of
  a two-dimensional array must have the same number of columns.
  
When these conditions are met, NumPy exploits these characteristics to
make the array faster, more memory efficient, and more convenient to use than
less restrictive data structures.
```

Reply:
👏 this is great!

Reply:
Agreed - LGTM!

Reply:
> Can I get them to render more compactly and to show all the grid lines?

Perhaps a literal block instead? I think it's worth avoiding markup tables where possible. No strong opinion though, we can always try different formats!

Reply:
Hmm, I thought about that, but I don't think I'll be happy with how it looks. 
I could try to do it in an equation, maybe, with LaTeX? Would that be ok?